### DISCUSSION TITLE
Customizable directories by OS?

### DISCUSSION THREAD

**Discussion by drmohundro** [twpayne/chezmoi#1433] [Q&A] [answered]
The scenario I'm thinking about here is the Neovim config... on Mac and Linux, Neovim follows the typical XDG approach and defaults to `~/.config/nvim`; however, on Windows, the default for configuration is under `~/AppData/Local/nvim`. My initial thought is to have a directory for Neovim config that has a per-OS customizable target (e.g. Mac/Linux => `~/.config/nvim` and Windows => `~/AppData/Local/nvim`), but maybe that isn't the best idea. I've read over the docs and searched through the issues, but I haven't found anything just yet.

I suppose another option might be to define the `$XDG_CONFIG_HOME` environment variable in Windows to point to `~/.config` and just use that along with a `dot_config` directory?

Just trying to weigh my options here.

Thanks!

**Comment by twpayne** [comment, marked as answer]
Symbolic links are one option, another is to use a shared template, as [described here](https://github.com/twpayne/chezmoi/blob/master/docs/HOWTO.md#handle-different-file-locations-on-different-systems-with-the-same-contents).

**Reply by drmohundro** [reply]
Wow, well I totally missed that - my apologies! Thanks for the response!

**Reply by LeonardoMor** [reply]
That link is now going nowhere

**Reply by twpayne** [reply]
Now at https://www.chezmoi.io/user-guide/manage-machine-to-machine-differences/#handle-different-file-locations-on-different-systems-with-the-same-contents.

**Reply by LeonardoMor** [reply]
Yeah found that later and came back to post the link.

For something like Neovim though, what's common is the entire `nvim` directory. I suppose in this case it would be better to use a symbolic link. Then edit `.chezmoiignore` accordingly.

Am I too far off?

**Reply by twpayne** [reply]
What you suggest makes sense.

In the long term I would like to implement https://github.com/twpayne/chezmoi/issues/2273, but I don't know when or if this will be done.

**Reply by LeonardoMor** [reply]
Yeah, it does look complicated.

The good thing is, for single OS use, chezmoi is pretty much there.

**Reply by LeonardoMor** [reply]
What does chezmoi do when the only thing inside a source dir is a simlink to another dir in the source dir?

**Comment by Archie-2021** [comment]
I have to use Windows , and like to use the same git repo I have for Linux. I don't get what should I do? like where should I have to clone my repo for chezmoi to look? `AppData/chezmoi` ? 
And should I have to change all my config files into templates? That would be a pain. I just want to share couple of my dotfiles, not much. and I also like to keep my ahk scripts inside my dotfiles and that is it. Can I do that without a major overhaul of my dotfiles repo?

**Reply by LeonardoMor** [reply]
Not necessarily. This is what I planned to do for example for Neovim, just have a directory that contains the Neovim configs. Then two more directories one for Linux and one for Windows, with the directory structure that each needs.

Then do symbolic links from the dir that contains the configs to the system dirs.

Then make it so that the Windows dir is ignored on Linux and viceversa.

Notice you can do it with 2 dirs instead of 3 :)

Repeat with any other multi OS config.

I didn´t get to do or test this as my employer finally gave me a Linux laptop so I don´t know if something like this is going to work.

**Reply by Triavanicus** [reply]
I am probably going to look into this, as I have to use Windows for school. Windows can create symbolic links, so maybe I could do something like create a file in chezmoi like `run_onchange_windows_create_links.ps1.tmpl` which then test to see if it is on Windows, and runs `mklink -D $env:USERPROFILE/.config/nvim $env:APPDATA/nvim` or something.

**Reply by tpsteiner** [reply]
Related: I use windows with WSL. AltSnap has been frustrating since the tool expects config file AltSnap.ini to be in %APPDATA%/AltSnap with no option to change the location. I created an absolute-path symlink in the appdata folder to the source file, which is in my WSL environment \\wsl$\Arch\home\[USER]\.local\share\chezmoi\dot_config\altsnap\AltSnap.ini
The symlink doesn't work, unfortunately. Maybe altsnap doesn't like the WSL path? I'm not sure what the solution is here, so I will just keep it out of my dotfiles :(

**Reply by LeonardoMor** [reply]
Is AltSnap one application that you use both on Windows and in WSL/Linux?

**Reply by tpsteiner** [reply]
I only use [AltSnap](https://github.com/RamonUnch/AltSnap/) on windows
EDIT: Some more context. I started with one dotfiles folder per system (one for Windows, another for WSL). Later I moved everything to the WSL environment to reduce updating steps, since they are both on the same hard drive anyways.

**Reply by LeonardoMor** [reply]
I see. In that case, I'd still clone the dotfiles repo on both WSL and Windows itself. Then on the `.chezmoiignore` file, I'd add something like:
```
{{- if eq .chezmoi.os "windows" }}
*
!path_to_AltSnap_config_relative_to_home
{{- end }}
```

But this does not avoid having the files duplicated in the system

**Reply by tpsteiner** [reply]
Thanks, that is a good way to solve it that does not introduce too much clutter. However I decided to solve it differently. I made a file run_after_sync-altsnap-to-windows.sh.tmpl to copy the files to windows after apply. That way i don't have to open a windows shell, then chezmoi update. I did this with my other windows config files as well, and it seems to work as expected.
